spring-integration-core/src/main/java/org/springframework/integration/mapping/AbstractHeaderMapper.java

```diff
@@ -388,31 +409,41 @@ private boolean containsIgnoreCase(String name) {
 			return false;
 		}
 
+		@Override
+		public boolean isNegated() {
+			return false;
+		}
+
 	}
 
 	/**
 	 * A pattern-based {@link HeaderMatcher} that matches if the specified
-	 * header match one of the specified simple patterns.
+	 * header matches one of the specified simple patterns.
 	 * @see org.springframework.util.PatternMatchUtils#simpleMatch(String, String)
 	 * @since 4.1
+	 * @deprecated since 4.3 - use a {@link CompositeHeaderMatcher} with
+	 * {@link SinglePatternBasedHeaderMatcher}s instead.
 	 */
+	@Deprecated
```

Comment:
No, I don't think that we should deprecate it.
Someone really may use it as is and will find arguments to reinstate it.
Anyway it doesn't hurt our code even if we don't use it anymore.